### PR TITLE
feat(Table): add virtual scroll in table filters tree

### PR DESCRIPTION
**Pull request by Ylg12345**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/41863

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   when the `filterMode` is set to `'tree'`, virtual scrolling is enabled by default, and the height of the virtual scrolling container is set to `264`.      |
| 🇨🇳 Chinese |   设置 `filterMode` 为 `'tree'` 时默认开启虚拟滚动，虚拟滚动容器高度为 `264`。    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 2f0fe97</samp>

This pull request enhances the table component by adding virtual scrolling and height customization for the filter menu. It modifies the `FilterDropdown` and `FilterMenu` components, the `ColumnType` interface, and the table documentation in both languages. It also adds a new demo file to show the usage of the new features.

### <samp>🤖 Generated by Copilot at 2f0fe97</samp>

*  Add the `height` and `virtual` props to the table column type, filter dropdown function, and filter menu component to enable and configure virtual scrolling in the filter menu with tree mode ([link](https://github.com/ant-design/ant-design/pull/42129/files?diff=unified&w=0#diff-af396bfff8e0daa4eb8b9f25ed47a6006b544d36c1733636ad578f7109f9ee5aR147-R148), [link](https://github.com/ant-design/ant-design/pull/42129/files?diff=unified&w=0#diff-af396bfff8e0daa4eb8b9f25ed47a6006b544d36c1733636ad578f7109f9ee5aR409-R410), [link](https://github.com/ant-design/ant-design/pull/42129/files?diff=unified&w=0#diff-3fee134a410a783dfd5f6f89b1a8f7e7eb1ba8e332aeaf4cafdefc4c723088e9R146-R149)).

### CODE REVIEW

File: components/table/demo/filter-virtual-in-tree.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## zh-CN
+
+使用 `height` 属性则切换为虚拟滚动，`height` 可设置的最大值为 264，因为 table 过滤下拉菜单的最大高度为 264。
```
Comment (afc163) [inline components/table/demo/filter-virtual-in-tree.md:3]: 这个 height 是否有必要对用户开放？

Comment (Ylg12345) [reply components/table/demo/filter-virtual-in-tree.md:3]: > 这个 height 是否有必要对用户开放？

这个height是传给Tree组件的，我翻了Tree组件的文档和代码, 它们是需要height开启虚拟化滚动的，或者我做内部判断，让用户传virtual来判断是否开启虚拟化滚动，当有virtual在内部设置一个height?

Comment (afc163) [reply components/table/demo/filter-virtual-in-tree.md:3]: 默认开启 virtual 会有啥问题不？感觉这两个属性都不用对用户开放。

Comment (Ylg12345) [reply components/table/demo/filter-virtual-in-tree.md:3]: 默认开启 virtual 看不出有什么问题，本质上是开启 Tree 组件的虚拟化滚动，对 table 组件没啥影响

Comment (Ylg12345) [reply components/table/demo/filter-virtual-in-tree.md:3]: > 默认开启 virtual 会有啥问题不？感觉这两个属性都不用对用户开放。

默认开启 virtual 看不出有什么问题，本质上是开启 Tree 组件的虚拟化滚动，对 table 组件没啥影响

Comment (afc163) [reply components/table/demo/filter-virtual-in-tree.md:3]: 那就先不开放这两个 prop，有需求再说。

Comment (Ylg12345) [reply components/table/demo/filter-virtual-in-tree.md:3]: okay, 那我直接在内部加吧，这边对 height 在内部也需要加个固定高度，你看高度多少比较好呢

Comment (afc163) [reply components/table/demo/filter-virtual-in-tree.md:3]: 264 如何

Comment (Ylg12345) [reply components/table/demo/filter-virtual-in-tree.md:3]: 可以的，那我改一下

Comment (Ylg12345) [reply components/table/demo/filter-virtual-in-tree.md]: 我改好啦，大佬有空可以看一下

File: components/table/demo/filter-in-tree.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
 ## zh-CN
 
-可以使用 `filterMode` 来修改筛选菜单的 UI，可选值有 `menu`（默认）和 `tree`。
+可以使用 `filterMode` 来修改筛选菜单的 UI，可选值有 `menu`（默认）和 `tree`，设置`filterMode` 为 `'tree'` 时默认开启虚拟滚动，虚拟滚动容器高度为 `264`。
```
Comment (afc163) [inline components/table/demo/filter-in-tree.md:3]:
```suggestion
可以使用 `filterMode` 来修改筛选菜单的 UI，可选值有 `menu`（默认）和 `tree`，设置`filterMode` 为 `'tree'` 时默认开启虚拟滚动，虚拟滚动容器高度为 `264px`。
```

File: components/table/hooks/useFilter/FilterDropdown.tsx [hunk 3]
```diff
@@ -404,6 +404,8 @@ function FilterDropdown<RecordType>(props: FilterDropdownProps<RecordType>) {
                       }
                     : undefined
                 }
+                height={264}
```
[the remark this answers is not in the document]
Comment (Ylg12345) [reply components/table/hooks/useFilter/FilterDropdown.tsx:407]: > 看看能不能直接用
> 
> https://github.com/ant-design/ant-design/blob/e9db1071d64863223c2529add349e7140d5a1ee9/components/table/style/index.ts#L327
> 
> 这个变量。

似乎不好用，这个变量没有暴露出来，都是使用getXXXStyle从token中解构的，在组件中好像用不了

**Comment by afc163** [comment]
好像不太能直接解决 https://github.com/ant-design/ant-design/issues/41863 ，`filterMode="menu"` 的时候还是卡。

**Comment by Ylg12345** [comment]
> 好像不太能直接解决 #41863 ，`filterMode="menu"` 的时候还是卡。

我也注意到了，tree 组件本身做了虚拟化滚动，传参就行了，menu的话，不支持这个，可能要在 rc-menu 里面做虚拟化滚动，代码更改可能过大

**Comment by Ylg12345** [comment]
> 好像不太能直接解决 #41863 ，`filterMode="menu"` 的时候还是卡。

可以的话，我试试往 rc-menu 提一个增加虚拟滚动功能的 pr，api设计就和 tree 组件一样？

**Comment by afc163** [comment]
可以试试

**Pull request closed without merging**